primitives/src/deeper_node.rs

```diff
@@ -0,0 +1,24 @@
+pub trait NodeInterface<AccountId, BlockNumber> {
+    /// This function tells if the device has been offline for a day
+    fn get_onboard_time(account_id: &AccountId) -> Option<BlockNumber>;
+
+    /// This function tells if the device has ever been online
+    fn im_ever_online(account_id: &AccountId) -> bool;
+
+    /// This function returns how many eras the device has been offline
+    fn get_eras_offline(account_id: &AccountId) -> u32;
+}
+
+impl<AccountId, BlockNumber> NodeInterface<AccountId, BlockNumber> for () {
```

Comment:
This implementation is not used, consider removing it.
![image](https://user-images.githubusercontent.com/4054836/175448344-ce99f891-b7a0-4442-b53c-102bd724aec5.png)

Reply:
as abovee